staker/block_validator.go

```diff
@@ -346,27 +337,34 @@ func NewBlockValidator(
 			return nil, err
 		}
 		if validated != nil {
-			ret.lastValidGS = validated.GlobalState
-		} else {
-			legacyInfo, err := ret.legacyReadLastValidatedInfo()
+			ret.instance.SetLastValidatedGlobalState(validated.GlobalState)
+		} else if ret.legacyInstance != nil {
+			legacyInfo, err := ret.legacyInstance.LegacyLastValidatedInfo()
 			if err != nil {
 				return nil, err
 			}
-			ret.legacyValidInfo = legacyInfo
+			ret.legacyInstance.SetLegacyValidInfo(legacyInfo)
 		}
 	}
 	// genesis block is impossible to validate unless genesis state is empty
-	if ret.lastValidGS.Batch == 0 && ret.legacyValidInfo == nil {
+	var legacy *legacyLastBlockValidatedDbInfo
+	if ret.legacyInstance != nil {
+		legacy, err = ret.legacyInstance.LegacyLastValidatedInfo()
```

Comment:
Wouldn't we have *just* looked this up on line 342?
Maybe we should capture it in a variable in the scope of the method to check it?
Oh, or is the idea that even if ResetBlockValidation is true, we might still want to check if there is a legacy validated db info? I'm a little confused by this code.